tests/unit/test.program.js

```diff
@@ -329,15 +331,26 @@ describe('program.main', () => {
 
 
 describe('program.defaultVersionGetter', () => {
+  let root = path.join(__dirname, '..', '..');
 
-  it('returns the package version', () => {
-    let root = path.join(__dirname, '..', '..');
+  it('returns the package version in production', () => {
     let pkgFile = path.join(root, 'package.json');
     return fs.readFile(pkgFile)
-      .then((pkgData) => {
-        assert.equal(defaultVersionGetter(root),
-                     JSON.parse(pkgData).version);
-      });
+    .then((pkgData) => {
+      const testProcess = {
+        env: {
+          NODE_ENV: 'production',
+        },
+      };
+      assert.equal(defaultVersionGetter(root, testProcess),
+                   JSON.parse(pkgData).version);
+    });
+  });
+
+  it('returns git commit information in development', () => {
+    const commit = `${git.branch()}-${git.long()}`;
+    assert.equal(defaultVersionGetter(root),
```

Comment:
This should be called using a `testProcess` object just like before. Whenever you have a test like this that relies on a specific state of a dependency (i.e. `process`) you need to explicitly set up that state, even though it's supposed to be the default state. Something like this:
````javascript
const testProcess = {
  // Empty env to simulate not declaring NODE_ENV.
  env: {},
};
defaultVersionGetter(root, testProcess)
````